pkg/planner/core/planbuilder.go

```diff
@@ -5942,6 +5943,37 @@ func checkAlterDDLJobOptValue(opt *AlterDDLJobOpt) error {
 			return fmt.Errorf("the value %v for %s is out of range [%v, %v]",
 				bs, opt.Name, variable.MinDDLReorgBatchSize, variable.MaxDDLReorgBatchSize)
 		}
+	case AlterDDLJobMaxWriteSpeed:
+		speed, err := GetMaxWriteSpeedNumericValFromExpression(opt)
+		if err != nil {
+			return err
+		}
+		if speed < 0 || speed > units.PiB {
+			return fmt.Errorf("the value %s for %s is out of range [%v, %v]",
+				strconv.FormatInt(speed, 10), opt.Name, 0, units.PiB)
+		}
 	}
 	return nil
 }
+
+// GetMaxWriteSpeedNumericValFromExpression gets the numeric value of the max write speed from the expression.
+func GetMaxWriteSpeedNumericValFromExpression(opt *AlterDDLJobOpt) (int64, error) {
+	var (
+		speed int64
+		err   error
+	)
+	v := opt.Value.(*expression.Constant)
```

Comment:
add unit tests to make sure it will always be a constant. For example, ALTER ... max_write_speed = RAND()